addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd

```diff
@@ -379,27 +383,111 @@ func is_angle_in_interval(angle: float, direction_angle: ESCDirectionAngle) -> b
+		last_dir = _get_dir_deg(deg, parent.animations)
+
+		# The character may have a state animation from before, which would be
+		# resumed, so we immediately force the correct idle animation
+		if parent.animation_sprite.animation != \
+				parent.animations.idles[last_dir].animation:
+			parent.animation_sprite.play(
+				parent.animations.idles[last_dir].animation
+			)
+		pose_scale = -1 if parent.animations.idles[last_dir].mirrored else 1
+	else:
+		var current_dir = last_dir
+		last_deg = deg
+		var target_dir = _get_dir_deg(deg, parent.animations)
+		
+		if target_dir - current_dir > 1:
+			var way_to_turn = get_shortest_way_to_dir(current_dir, target_dir)
+		
+			var dir = current_dir
+			while dir != target_dir:
+				dir += way_to_turn
+				if dir >= parent.animations.dir_angles.size():
+					dir = 0
+				if dir < 0:
+					dir = parent.animations.dir_angles.size() - 1
+				
+#				if parent.animation_sprite.animation != \
+#						parent.animations.idles[last_dir].animation:
+				parent.animation_sprite.play(
+					parent.animations.idles[dir].animation
+				)
+				yield(parent.animation_sprite, "animation_finished")
+				pose_scale = -1 if parent.animations.idles[dir].mirrored else 1
+			
+				
+			
+		else:
+			# The character may have a state animation from before, which would be
+			# resumed, so we immediately force the correct idle animation
+			if parent.animation_sprite.animation != \
+					parent.animations.idles[last_dir].animation:
+				parent.animation_sprite.play(
+					parent.animations.idles[last_dir].animation
+				)
+			pose_scale = -1 if parent.animations.idles[last_dir].mirrored else 1
+		
 	update_terrain()
 
+
 # Returns the angle that corresponds to the current direction of the object.
 func _get_angle() -> int:
 	return parent.animations.dir_angles[last_dir].animation
+
 	
+# Return the shortest way to turn from a direction to another. Returned way is
+# either: 
+# -1 (shortest way is to turn anti-clockwise) 
+# 0 (already at the right direction)
+# 1 (clockwise).
+#
+# ####Parameters
+# - current_dir: integer corresponding to the starting direction as defined in
+# the attached ESCAnimationResource.directions.
+# - target_dir: integer corresponding to the target direction as defined in
+# the attached ESCAnimationResource.directions.
+#
+# *Returns*
+# Integer: -1 (anti-clockwise), 1 (clockwise) or 0 (no movement needed).
+func get_shortest_way_to_dir(current_dir: int, target_dir: int) -> int:
+	if current_dir < 0 or current_dir > parent.animations.dir_angles.size() - 1:
+		escoria.logger.report_errors(
+			"esc_movable.gd:get_shortest_way_to_dir()",
+			["Invalid direction (current_dir) " + str(current_dir)]
+		)
+	
+	if target_dir < 0 or target_dir > parent.animations.dir_angles.size() - 1:
+		escoria.logger.report_errors(
+			"esc_movable.gd:get_shortest_way_to_dir()",
+			["Invalid direction (target_dir) " + str(target_dir)]
+		)
+	
+	if current_dir == target_dir:
+		return 0
+	
+	var internal = false
```

Comment:
What does that variable do? I don't understand its purpose.

Reply:
If internal=true, then the angle rotation finishes in the [0;7] interval and does not need to pass down 0 (back to 7) or upper than 7 (back to 0).

Reply:
Ah okay. That makes sense then. Thanks